src/UniV4MerklRewards.sol

```diff
@@ -0,0 +1,158 @@
+                IS_OUT_OF_RANGE_INCENTIVIZED,
+                WEIGHT_FEES,
+                WEIGHT_TOKEN_0,
+                WEIGHT_TOKEN_1,
+                480, // 15 * 32, offset for whitelist address
+                512, // 16 * 32, offset for blacklist address
+                576 // 18 * 32, offset for hooks
+            ),
+            abi.encode(
+                0, // lowerPriceTolerance
+                0, // upperPriceTolerance
+                0, // lowerPriceBound
+                0, // upperPriceBound
+                608, // 19 * 32, offset for empty unknown last param
+                32, // poolId len as bytes
+                UNIV4_POOL_ID,
+                0, // empty whitelist
+                1, // blacklist len
+                LIQUITY_FUNDS_SAFE, // blacklisted address
+                0, // empty hooks
+                0 // empty last unknown param
+            )
+        );
+    }
+
+    function onRegisterInitiative(uint256 _atEpoch) external override {}
+
+    /// @notice Callback hook that is called by Governance after the initiative was unregistered
+    /// @param _atEpoch Epoch at which the initiative is unregistered
+    function onUnregisterInitiative(uint256 _atEpoch) external override {}
+
+    /// @notice Callback hook that is called by Governance after the LQTY allocation is updated by a user
+    /// @param _currentEpoch Epoch at which the LQTY allocation is updated
+    /// @param _user Address of the user that updated their LQTY allocation
+    /// @param _userState User state
+    /// @param _allocation Allocation state from user to initiative
+    /// @param _initiativeState Initiative state
+    function onAfterAllocateLQTY(
+        uint256 _currentEpoch,
+        address _user,
+        IGovernance.UserState calldata _userState,
+        IGovernance.Allocation calldata _allocation,
+        IGovernance.InitiativeState calldata _initiativeState
+    ) external override {}
+
+    /// @notice Callback hook that is called by Governance after the claim for the last epoch was distributed
+    /// to the initiative
+    /// @param _claimEpoch Epoch at which the claim was distributed
+    /// @param _bold Amount of BOLD that was distributed
+    function onClaimForInitiative(uint256 _claimEpoch, uint256 _bold) external override onlyGovernance {}
+
+    function createCampaign(uint256 _amount) internal {
+        // Avoid if rewards too low
+        if (_amount < CAMPAIGN_BOLD_AMOUNT_THRESHOLD) return;
+
+        uint256 claimEpoch = governance.epoch() - 1;
+
+        // (Only once per epoch)
+        uint256 epochEnd = EPOCH_START + claimEpoch * EPOCH_DURATION;
+        IDistributionCreator.CampaignParameters memory params = IDistributionCreator.CampaignParameters({
+            campaignId: bytes32(0),
+            creator: address(this),
+            rewardToken: address(boldToken),
+            amount: _amount,
+            campaignType: CAMPAIGN_TYPE,
+            startTimestamp: uint32(epochEnd),
```

Comment:
So the intent is to start the distribution at the end of the epoch, then have it last for one more week thereafter right?

Reply:
Yes.